src/hotspot/share/memory/allocation.hpp

```diff
@@ -443,20 +441,10 @@ class ResourceObj ALLOCATION_SUPER_CLASS_SPEC {
       return res;
   }
 
-  void* operator new [](size_t size) throw() {
-      address res = (address)resource_allocate_bytes(size);
-      DEBUG_ONLY(set_allocation_type(res, RESOURCE_AREA);)
-      return res;
-  }
-
-  void* operator new [](size_t size, const std::nothrow_t& nothrow_constant) throw() {
-      address res = (address)resource_allocate_bytes(size, AllocFailStrategy::RETURN_NULL);
-      DEBUG_ONLY(if (res != NULL) set_allocation_type(res, RESOURCE_AREA);)
-      return res;
-  }
-
+  void* operator new [](size_t size) throw() = delete;
```

Comment:
I think it should suffice to `delete` this one, with the others completely removed.  I think that name lookup will find this, and one will either get an error because the arguments don't match or because they do match to a deleted function.

Reply:
Sorry for the delay, can I keep all `= delete`? I have an idea how to fix them in the future.

Reply:
Sure.